### PR TITLE
Catalog API should support tokens

### PR DESCRIPTION
**Pull request by racktear**
In a case where the acl_token is set to "anonymous" in consul server config, by default services in catalog are invisible, and trying to list them shows only "consul" service.

In context of #116

### CODE REVIEW

**Comment by gregdurham** [comment]
I believe this is incorrect for the consul API:
https://www.consul.io/docs/agent/http/catalog.html
namely:

> An optional ACL token may be provided to perform the registration by including a WriteRequest block in the query payload, like this:
{
  "WriteRequest": {
    "Token": "foo"
  }
}

**Comment by gregdurham** [comment]
`          token = token or self.agent.token
            if token:
                data['WriteRequest'] = {}
                data['WriteRequest']['Token'] = token`

File: consul/base.py [hunk 1]
```diff
@@ -994,10 +1002,18 @@ def register(self, node, address, service=None, check=None, dc=None):
                 data['service'] = service
             if check:
                 data['check'] = check
+            token = token or self.agent.token
+            if token:
+                data['token'] = token
```
Comment (gregdurham) [inline consul/base.py]: should be:
data['WriteRequest'] = {}
data['WriteRequest']['Token'] = token

File: consul/base.py [hunk 2]
```diff
@@ -1020,6 +1038,9 @@ def deregister(self, node, service_id=None, check_id=None, dc=None):
                 data['serviceid'] = service_id
             if check_id:
                 data['checkid'] = check_id
+            token = token or self.agent.token
+            if token:
+                data['token'] = token
```
Comment (gregdurham) [inline consul/base.py]: should be:
data['WriteRequest'] = {}
data['WriteRequest']['Token'] = token

**Comment by racktear** [comment]
@gregdurham thanks. Updated.

**Comment by gregdurham** [comment]
@racktear Thank you for getting this tied in!

**Comment by racktear** [comment]
Hello!

Just wanted to follow up on this pull request. Could you please merge it?

**Comment by racktear** [comment]
@joshainglis Hi, could you please merge this PR? It is required to fully support ACL tokens.

**Comment by cablehead** [comment]
Hi Konstantin. As always apologies for letting these PR sit idle for so
long. I'm going to set this Saturday aside to do some triaging to try and
catch back up. At the same time I'll try and identify more people to bring
on as maintainers.

**Pull request merged**